java/grpc-client/src/test/java/com/youtube/vitess/client/grpc/GrpcClientTlsTest.java

```diff
@@ -0,0 +1,203 @@
+        ).start();
+        createKey.waitFor();
+
+        final Process createCert = new ProcessBuilder(
+                // openssl req -new -x509 -nodes -days 3600 -batch -config ca.config -key ca-key.pem -out ca-cert.pem
+                Arrays.asList(
+                        "openssl",
+                        "req",
+                        "-new",
+                        "-x509",
+                        "-nodes",
+                        "-days",
+                        "3600",
+                        "-batch",
+                        "-config",
+                        certDirectory.getCanonicalPath() + File.separatorChar + "ca.config",
+                        "-key",
+                        certDirectory.getCanonicalPath() + File.separatorChar + "ca-key.pem",
+                        "-out",
+                        certDirectory.getCanonicalPath() + File.separatorChar + "ca-cert.pem")
+        ).start();
+        createCert.waitFor();
+
+        // TODO: Create JKS trustStore
+    }
+
+    private static void createSignedCert(final String ca, final String serial, final String name, final String commonName) throws Exception {
+        final String config = certDirectory.getCanonicalPath() + File.separatorChar + name + ".config";
+        final String key = certDirectory.getCanonicalPath() + File.separatorChar + name + "-key.pem";
+        final String req = certDirectory.getCanonicalPath() + File.separatorChar + name + "-req.pem";
+        final String cert = certDirectory.getCanonicalPath() + File.separatorChar + name + "-cert.pem";
+        final String caCert = certDirectory.getCanonicalPath() + File.separatorChar + ca + "-cert.pem";
+        final String caKey = certDirectory.getCanonicalPath() + File.separatorChar + ca + "-key.pem";
+
+        final StringWriter writer = new StringWriter();
+        IOUtils.copy(GrpcClientTlsTest.class.getResourceAsStream("/cert-template.config"), writer, "UTF-8");
+        final String content = writer.toString();
+        content.replaceAll("%s", commonName);
+        java.nio.file.Files.write(Paths.get(config), content.getBytes("UTF-8"));
+
+        final Process createKeyAndCSR = new ProcessBuilder(
+                // openssl req -newkey rsa:2048 -days 3600 -nodes -batch -config cert-template.config -keyout vtgate-client-key.pem -out vtgate-client-req.pem
+                Arrays.asList(
+                        "openssl",
+                        "req",
+                        "-newkey",
+                        "rsa:2048",
+                        "-days",
+                        "3600",
+                        "-nodes",
+                        "-batch",
+                        "-config",
+                        config,
+                        "-keyout",
+                        key,
+                        "-out",
+                        req)
+        ).start();
+        createKeyAndCSR.waitFor();
+
+        final Process askHarshit = new ProcessBuilder(
```

Comment:
This `askHarshi` step is taken straight from the `encrypted_transport.py` test script (line 102), but I don't really understand what it's doing.  The `-in` and `-out` parameters are the same filename.